sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -307,4 +317,186 @@ public PlannerHook hook()
       return hook;
     }
   }
+
+  public static DruidException translateException(Exception e)
+  {
+    try {
+      throw e;
+    }
```

Comment:
any reason not to just make all of these `instanceof` checks, especially since some of the catch still have `instanceof`?

Reply:
not really, I think I started doing that and then stopped for some reason, I don't remember why though.